src/test/extension.test.ts

```diff
@@ -1,110 +1,211 @@
-            hoverMessage: ' Please fill out the following configurations in your Sentry extension settings: repoMatch',
-            linkURL:
-                'https://sentry.io/organizations/sourcegraph/issues/?project=134412&query=is%3Aunresolved+cannot%20determine%20file%20path&statsPeriod=14d',
-        },
-    },
-    {
-        range: new sourcegraph.Range(1, 0, 1, 0),
-        isWholeLine: true,
-        after: {
-            backgroundColor: '#f2736d',
-            color: 'rgba(255, 255, 255, 0.8)',
-            contentText: ' View logs in Sentry (❕)» ',
-            hoverMessage:
-                ' Please fill out the following configurations in your Sentry extension settings: repoMatch, fileMatch',
-            linkURL:
-                'https://sentry.io/organizations/sourcegraph/issues/?project=134412&query=is%3Aunresolved+cannot%20determine%20file%20path&statsPeriod=14d',
+        expected: {
+            range: new sourcegraph.Range(new sourcegraph.Position(1, 0), new sourcegraph.Position(1, 0)),
+            isWholeLine: true,
+            after: {
+                backgroundColor: '#f2736d',
+                color: 'rgba(255, 255, 255, 0.8)',
+                contentText: ' View logs in Sentry (❕)» ',
+                hoverMessage: ' Add Sentry projects to your Sentry extension settings for project matching.',
+                linkURL: 'https://sentry.io/organizations/sourcegraph/issues/',
+            },
         },
     },
     {
-        range: new sourcegraph.Range(1, 0, 1, 0),
-        isWholeLine: true,
-        after: {
-            backgroundColor: '#f2736d',
-            color: 'rgba(255, 255, 255, 0.8)',
-            contentText: ' View logs in Sentry (❕)» ',
-            hoverMessage: ' Add Sentry projects to your Sentry extension settings for project matching.',
-            linkURL: 'https://sentry.io/organizations/sourcegraph/issues/',
+        goal:
+            'match line based on common pattern, render warning link hinting to add projectId and render link to general issues page',
+        index: 1,
+        match: '',
+        missingConfigData: ['repoMatches', 'fileMatches'],
+        sentryProjectId: undefined,
+        expected: {
+            range: new sourcegraph.Range(new sourcegraph.Position(1, 0), new sourcegraph.Position(1, 0)),
+            isWholeLine: true,
+            after: {
+                backgroundColor: '#f2736d',
+                color: 'rgba(255, 255, 255, 0.8)',
+                contentText: ' View logs in Sentry (❕)» ',
+                hoverMessage: ' Add Sentry projects to your Sentry extension settings for project matching.',
+                linkURL: 'https://sentry.io/organizations/sourcegraph/issues/',
+            },
         },
     },
 ]
 
 describe('decorate line', () => {
-    for (const [i, deco] of data.entries()) {
+    sourcegraph.configuration.get().update('sentry.organization', 'sourcegraph')
+    sourcegraph.configuration.get().update('sentry.projects', projects)
```

Comment:
"Promises must be handled appropriately" is emitted for this line because `update` returns a Promise that is not being `await`ed or chained to another Promise.

These lines look like initialization, which could be moved into an async `beforeEach` hook like so:

```typescript
describe(... () => {
  beforeEach(async () => { config updates... })
  it(...)
})
```

Reply:
I still get the tslint error using beforeEach, but if I void the responses, e.g. `void sourcegraph.configuration.get().update('sentry.projects', projects)`, it works.

Reply:
@vanesa when moving it to a `beforeEach()`, you also need to use the `await` operator